packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -0,0 +1,1274 @@
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     key: scaffoldKey,
+///     body: Center(
+///       child: Container(
+///         width: 100,
+///         height: 100,
+///         child: CupertinoContextMenu(
+///           child: Container(
+///             color: Colors.red,
+///           ),
+///           actions: <CupertinoContextMenuAction>[
+///             CupertinoContextMenuAction(
+///               child: const Text('Action one'),
+///               onPressed: () {
+///                 Navigator.pop(context);
+///               },
+///             ),
+///             CupertinoContextMenuAction(
+///               child: const Text('Action two'),
+///               onPressed: () {
+///                 Navigator.pop(context);
+///               },
+///             ),
+///           ],
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+///   * [Apple's HIG for Context Menus](https://developer.apple.com/design/human-interface-guidelines/ios/controls/context-menus/)
+class CupertinoContextMenu extends StatefulWidget {
+  /// Create a context menu.
+  ///
+  /// [actions] is required and cannot be null. It must contain at least one
+  /// [CupertinoContextMenuAction].
+  ///
+  /// [child] is required and cannot be null.
+  CupertinoContextMenu({
+    Key key,
+    @required this.actions,
+    @required this.child,
+    this.previewBuilder,
+  }) : assert(actions != null && actions.isNotEmpty),
+       assert(child != null),
+       super(key: key);
+
+  /// The widget that can be "opened" with the [CupertinoContextMenu].
+  ///
+  /// When the [CupertinoContextMenu] is long-pressed, the menu will open and
+  /// this widget (or the widget returned by [previewBuilder], if provided) will
+  /// be moved to the [Overlay] and placed inside of an [Expanded] widget. This
+  /// allows the child to resize to fit in the full-screen Overlay, if it
+  /// doesn't size itself.
+  ///
+  /// When the [CupertinoContextMenu] is "closed", this widget acts like a
+  /// [Container], i.e. it does not constrain its child's size or affect its
+  /// position.
+  ///
+  /// This parameter cannot be null.
+  final Widget child;
+
+  /// The actions that are shown in the menu.
+  ///
+  /// This parameter cannot be null, and items in the List must be
+  /// [CupertinoContextMenuAction]s.
+  final List<CupertinoContextMenuAction> actions;
+
+  /// A function that returns an alternative widget to show when the
+  /// [CupertinoContextMenu] is open.
+  ///
+  /// If not specified, [child] will be shown.
+  ///
+  /// The preview is often used to show a slight variation of the [child]. For
+  /// example, the child could be given rounded corners in the preview but have
+  /// sharp corners when in the page.
+  ///
+  /// In addition to the current [BuildContext], the function is also called
+  /// with an [Animation] and the [child]. The animation is used internally to
```

Comment:
instead of "is used internally", you could explain that the animation goes from 0-1 when the menu opens, 1-0 when it closes.